webapp/src/app/course-welcome/course-welcome.component.ts

```
@@ -140,6 +210,295 @@ export class CourseWelcomeComponent implements OnInit {
+      // Start reading the file
+      reader.readAsDataURL(this.chosenFile);
+
+      return newFileName;
+    } else {
+      // If no file is uploaded, fetch and upload a random image
+      return null;
+    }
+  }
+  extractTimestamp(url: string): number | null {
+    const timestampMatch = url.match(/t=(\d+)/);
+    return timestampMatch ? parseInt(timestampMatch[1], 10) : null;
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)